allennlp/training/deepspeed/utils.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from deepspeed.utils import logger as ds_logger
+ds_logger.setLevel(logging.WARNING)
+ds_logger.propagate = False
+
+import torch
+from allennlp.models.model import Model
+from allennlp.common import Lazy
+from allennlp.common.checks import ConfigurationError
+from allennlp.training.deepspeed.config import DeepspeedConfig, DeepspeedArgs
+
+import deepspeed
+from deepspeed.runtime.engine import DeepSpeedEngine
+
+def launch_deepspeed(
+    model: Model,
+    optimizer: torch.optim.Optimizer,
+    config: DeepspeedConfig,
+    args: Lazy[DeepspeedArgs],
+    batch_size: int,
+    gradient_accumulation_steps: int,
+):
+    if not(optimizer is None or config.optimizer is None):
+        raise ConfigurationError(f"Cannot provide both optimizer and deepspeed_optimizer. {optimizer, config.to_dict()}")
+
+    config = dict(**{k: v for k, v in config.to_dict().items() if v is not None}, train_batch_size=batch_size, gradient_accumulation_steps=gradient_accumulation_steps)
+    ds = DeepSpeedEngine(
+        args=args,
+        model=model,
+        optimizer=optimizer,
+        model_parameters=model.parameters(),
+        dist_init_required=False,
+        config_params=config
+    )
+    if hasattr(ds, 'timers'):
+        def mute_log(*args, **kwargs):
+            pass
+        ds.timers.log = mute_log
+    return ds, ds.optimizer
```

Comment:
Does this have to be in a separate function? It's only called in one place, so it could live there?

If we put it here, we are sort of creating a new public API. I'd like to avoid that if we don't have to.

Reply:
Yep, I'll just move this to the constructor.